frame/ibc/src/host_functions.rs

```diff
@@ -55,4 +51,60 @@ impl HostFunctionsProvider for HostFunctions {
 	fn sha256_digest(data: &[u8]) -> [u8; 32] {
 		sp_io::hashing::sha2_256(data)
 	}
+
+	// These are temporary implementations to fix build errors, will be replaced with
+	// host functions in https://github.com/ComposableFi/composable/pull/1122
+	fn ripemd160(message: &[u8]) -> [u8; 20] {
+		use ripemd::Digest;
+		let mut hasher = ripemd::Ripemd160::new();
+		hasher.update(message);
+		let hash = hasher.finalize();
+		let mut res = [0u8; 20];
+		res.copy_from_slice(&hash);
+		res
+	}
+
+	fn ed25519_verify(signature: &[u8; 64], msg: &[u8], pubkey: &[u8]) -> bool {
```

Comment:
I'd personally change the signature to be `ed25519_verify(signature: &[u8; 64], msg: &[u8], pubkey: &ed25519::Public) -> bool` . That way the caller can evaluate if it is a pubkey mismatch, or invalid pubkey. Now we hide that fact.

Reply:
The trait definition is specified in `ibc-rs`, will keep this in mind for the next update.

Reply:
Ah alright gotcha